src/org/labkey/test/util/ExperimentalFeaturesHelper.java

```diff
@@ -15,114 +15,103 @@
-            if (response.containsKey("success") && (Boolean)response.get("success"))
-            {
-                Map<String, Boolean> data = (Map<String, Boolean>)response.get("data");
-                return data;
-            }
-        }
-        catch (IOException e)
-        {
-            throw new RuntimeException("Failed to get experimental features", e);
-        }
-        catch (CommandException e)
+        if (_originalFeatureFlags.isEmpty())
+            return;
+
+        TestLogger.log("Resetting experimental flags to their original value:");
+
+        Map<String, Boolean> flags = new HashMap<>(_originalFeatureFlags);
+        _originalFeatureFlags = new HashMap<>();
+
+        Connection cn = test.createDefaultConnection(false);
+        for (Map.Entry<String, Boolean> features : flags.entrySet())
         {
-            TestLogger.log("Unable to get experimental features. Ignoring: " + e.getStatusCode());
+            setExperimentalFeature(cn, features.getKey(), features.getValue(), false);
         }
-
-        return emptyMap();
     }
 
-    // Set the current experimental flags
-    public static Map<String, Boolean> setExperimentalFeatures(Connection cn, Map<String, Boolean> flags)
+    private static void setExperimentalFeature(Connection cn, String feature, boolean enable, boolean savePrevious)
     {
-        PostCommand command = new PostCommand("admin", "setExperimentalFeatures.api");
-
-        // don't send apiVersion
-        command.setRequiredVersion(-1);
+        TestLogger.log((enable ? "Enabling" : "Disabling") + " experimental feature " + feature);
 
-        JSONObject json = new JSONObject();
-        json.put("flags", flags);
-        command.setJsonObject(json);
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("feature", feature);
+        parameters.put("enabled", enable);
 
+        PostCommand command = new PostCommand("admin", "experimentalFeature");
+        command.setParameters(parameters);
         try
         {
             CommandResponse r = command.execute(cn, null);
             Map<String, Object> response = r.getParsedData();
-            if (response.containsKey("success") && (Boolean)response.get("success"))
+            if (savePrevious && response.containsKey("previouslyEnabled"))
             {
-                Map<String, Boolean> data = (Map<String, Boolean>)response.get("data");
-                return data;
+                // Remember the previous setting for the feature flag if was different
+                Boolean previouslyEnabled = (Boolean)response.get("previouslyEnabled");
+                if (!Objects.equals(enable, previouslyEnabled.booleanValue()))
+                {
+                    TestLogger.log("Experimental feature will be reset back to " + (previouslyEnabled ? "enabled" : "disabled") + " after test is completed");
+                    _originalFeatureFlags.put(feature, previouslyEnabled.booleanValue());
```

Comment:
Remove the `savePrevious` flag and always save the previous value if there isn't an existing saved value. Then clear out `_originalFeatureFlags` at the end of `resetFeatures`.
That way, enabling/disabling a feature multiple times won't clobber the initial setting.